### PR TITLE
Use Gradle To Download Blockly Web Files & Add Other Lanuage Demos

### PR DESCRIPTION
**Pull request by TheBrokenRail**
This makes sure when you build ```blockly-android``` you will always get the latest stable version of Blockly. It also gives examples if you want to generate for other languages.

---
This change is [Reviewable](https://reviewable.io/reviews/google/blockly-android/757)

### CODE REVIEW

File: blocklylib-core/build.gradle [hunk 1]
```diff
@@ -43,6 +43,28 @@ task sourcesJar(type: Jar) {
...
+task getBlockly {
+    def files = [
+            'blockly_compressed.js': 'src/main/assets/blockly_compressed.js',
+            'blocks_compressed.js': 'src/main/assets/blocks_compressed.js',
+            'javascript_compressed.js': 'src/main/assets/generators/javascript_compressed.js',
+            'lua_compressed.js': 'src/main/assets/generators/lua_compressed.js',
+            'php_compressed.js': 'src/main/assets/generators/php_compressed.js',
+            'python_compressed.js': 'src/main/assets/generators/python_compressed.js',
+            'dart_compressed.js': 'src/main/assets/generators/dart_compressed.js',
+            'msg/js/en.js': 'src/main/assets/msg/js/en.js'
```
Comment (AnmAtAnm) [inline blocklylib-core/build.gradle]: This means every project will be including these other language files, yet very few will need them.

Can we move the non JavaScript generator files to the demo gradle file (and test project gradle file, it looks like), and add a comment in each of the demo activity classes to note the download via the gradle file?

Leave a comment in this file noting that the other languages are available.

File: blocklylib-core/src/main/java/com/google/blockly/android/codegen/LanguageDefinition.java [hunk 2]
```diff
@@ -11,7 +11,31 @@
      * Standard definition for the JavaScript language generator.
      */
     public final static LanguageDefinition JAVASCRIPT_LANGUAGE_DEFINITION
-            = new LanguageDefinition("javascript_compressed.js", "Blockly.JavaScript");
+            = new LanguageDefinition("generators/javascript_compressed.js", "Blockly.JavaScript");
+
+    /**
+     * Standard definition for the Lua language generator.
+     */
```
Comment (AnmAtAnm) [inline blocklylib-core/src/main/java/com/google/blockly/android/codegen/LanguageDefinition.java]: Following up with the comment on the `build.gradle`, include a not in this and following doc comments that these non-JavaScript files are not included in core, and need to be added to your project manually. ("See demo build.gradle for an example.")

File: blocklytest/src/androidTest/assets/generators/test_dart.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Generating Python for test blocks.
```
Comment (AnmAtAnm) [inline blocklytest/src/androidTest/assets/generators/test_dart.js]: Python -> Dart

File: blocklydemo/src/main/java/com/google/blockly/android/demo/PHPActivity.java [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+import com.google.blockly.android.codegen.LanguageDefinition;
+import com.google.blockly.model.DefaultBlocks;
+
+import java.util.Arrays;
+import java.util.List;
+
+
+/**
+ * Demo activity that programmatically adds a view to split the screen between the Blockly workspace
+ * and an arbitrary other view or fragment.
```
Comment (AnmAtAnm) [inline blocklydemo/src/main/java/com/google/blockly/android/demo/PHPActivity.java]: Update documentation.

File: blocklydemo/src/main/java/com/google/blockly/android/demo/PythonActivity.java [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+import com.google.blockly.android.codegen.LanguageDefinition;
+import com.google.blockly.model.DefaultBlocks;
+
+import java.util.Arrays;
+import java.util.List;
+
+
+/**
+ * Demo activity that programmatically adds a view to split the screen between the Blockly workspace
+ * and an arbitrary other view or fragment.
```
Comment (AnmAtAnm) [inline blocklydemo/src/main/java/com/google/blockly/android/demo/PythonActivity.java]: Update documentation.

File: blocklylib-core/build.gradle (hunk 1 above)
Comment (TheBrokenRail) [reply blocklylib-core/build.gradle]: Created Full and Lite Product Flavors.

File: blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGeneratorService.java [hunk 6]
```diff
@@ -85,7 +85,7 @@ public void onPageFinished(WebView view, String url) {
                 handleRequest();
             }
         });
-        mWebview.loadUrl(BLOCKLY_COMPILER_PAGE);
+        mWebview.loadUrl("about:blank");
```
[the remark this answers is not in the document]
Comment (TheBrokenRail) [reply blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGeneratorService.java]: Done!

File: blocklytest/src/androidTest/assets/generators/test_php.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Generating Python for test blocks.
```
[the remark this answers is not in the document]
Comment (TheBrokenRail) [reply blocklytest/src/androidTest/assets/generators/test_php.js]: Done!

File: blocklydemo/src/main/java/com/google/blockly/android/demo/DartActivity.java [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+import com.google.blockly.android.codegen.CodeGenerationRequest;
+import com.google.blockly.android.codegen.LanguageDefinition;
+import com.google.blockly.model.DefaultBlocks;
+
+import java.util.Arrays;
+import java.util.List;
+
+
+/**
+ * Demo activity that programmatically adds a view to split the screen between the Blockly workspace
```
[the remark this answers is not in the document]
Comment (TheBrokenRail) [reply blocklydemo/src/main/java/com/google/blockly/android/demo/DartActivity.java]: Done!

File: blocklydemo/src/main/java/com/google/blockly/android/demo/PHPActivity.java [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+        super.onClearWorkspace();
+        mGeneratedTextView.setText(mNoCodeText);
+        updateTextMinWidth();
+    }
+
+    /**
+     * Estimate the pixel size of the longest line of text, and set that to the TextView's minimum
+     * width.
+     */
+    private void updateTextMinWidth() {
```
[the remark this answers is not in the document]
Comment (TheBrokenRail) [reply blocklydemo/src/main/java/com/google/blockly/android/demo/PHPActivity.java]: Done!

File: blocklydemo/src/main/java/com/google/blockly/android/demo/PHPActivity.java (hunk 4 above)
Comment (TheBrokenRail) [reply blocklydemo/src/main/java/com/google/blockly/android/demo/PHPActivity.java]: Done!

File: blocklydemo/src/main/java/com/google/blockly/android/demo/PythonActivity.java (hunk 5 above)
Comment (TheBrokenRail) [reply blocklydemo/src/main/java/com/google/blockly/android/demo/PythonActivity.java]: Done!

**Comment by TheBrokenRail** [comment]
You can add ```:lite@aar``` or ```:full@aar``` to your dependency line for JitPack or JCenter, to select flavor or keep as is to use the full version.

**Comment by TheBrokenRail** [comment]
Done!

**Comment by TheBrokenRail** [comment]
Ready to approve?

**Comment by AnmAtAnm** [comment]
After merging this, I'm getting the following error when building `develop`:

```
FAILURE: Build failed with an exception.

* What went wrong:
Task 'assembleDebug' not found in project ':blocklydemo'.

* Try:
Run gradle tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
```

**Pull request merged**